### PR TITLE
[isoltest] Add account builtin.

### LINKED ISSUE #11310: [isoltest] Add option to chose sender account
Currently there is no way to change the sender account. A lot of tests (e.g. those in `test/contracts/*`) depend on the ability to call a contract with different senders.

A potential option (if the semantictest tool keeps a state) is to introduce a line such as `account: <num>`, where `num` is an index starting from 0 (the default), which changes the sender.

Example:
```
// account: 0
// reserve(string), 50 ether: 0x20, 3, "abc" ->
// owner(string): 0x20, 3, "abc" -> 0x1212121212121212121212121212120000000012
// account: 1
// reserve(string), 100 ether: 0x20, 3, "abc" ->
// owner(string): 0x20, 3, "abc" -> 0x1212121212121212121212121212120000000012
```

The `account: 0` can be omitted, as that is the starting value, however it can be used to go back to the initial account.

Alternative is to use the actual addresses instead of indexes, but I'm not sure we need that as long as the account addresses are deterministic.

### PR DESCRIPTION
**Pull request by aarlt**
Closes #11310.

### CODE REVIEW

File: test/ExecutionFramework.h [hunk 1]
```diff
@@ -241,6 +241,12 @@ class ExecutionFramework
 		return result;
 	}
 
+	util::h160 setAccount(size_t _accountNumber)
```
Comment (axic) [inline test/ExecutionFramework.h:244]: Hmm, how about `changeCurrentAccount` or `setCurrentSender` or something in that direction?

File: test/libsolidity/semanticTests/isoltestTesting/account.sol [hunk 2]
```diff
@@ -0,0 +1,48 @@
+contract AccountBuiltinTest {
+    function who_am_i() public returns (address result) {
+        result = msg.sender;
+    }
+}
+// ====
+// compileViaYul: also
+// ----
+// constructor()
+// account: 0 -> 0x1212121212121212121212121212120000000012
```
Comment (axic) [inline test/libsolidity/semanticTests/isoltestTesting/account.sol:10]: Hmm, actually displaying the resulting address is nice.

**Review by axic — DISMISSED** [review]
I think this looks great, I would still suggest to improve the name of `setAccount` and wait for some other review, but could be merged as-is imho.

File: test/libsolidity/SemanticTest.cpp [hunk 3]
```diff
@@ -151,9 +151,18 @@ map<string, Builtin> SemanticTest::makeBuiltins() const
...
 		 	}
-		}
+		},
+		{
+			"account",
+			[this](FunctionCall const& _call) -> optional<bytes>
+			{
+				soltestAssert(_call.arguments.parameters.size() == 1, "Account number expected.");
+				size_t accountNumber = static_cast<size_t>(stoi(_call.arguments.parameters.at(0).rawString));
+				return toBigEndian(u256(ExecutionFramework::setAccount(accountNumber).asBytes()));
```
Comment (axic) [inline test/libsolidity/SemanticTest.cpp]: I don't think you need to get it into u256 again after `asBytes` as that serialises already:
```suggestion
				return ExecutionFramework::setAccount(accountNumber).asBytes();
```

I think this is also the reason of the test failure as the expectation turns into 32-bytes, but you have 20-bytes, i.e. needs the leading zeroes.

Comment (axic) [reply test/libsolidity/SemanticTest.cpp]: So apparently `TestFunctionCall/BytesUtils` is rather broken and `BytesUtils::formatHex` will mess up the formatting. The proper solution (for now) is to right align this via:
```suggestion
				return toBigEndian(h256(ExecutionFramework::setAccount(accountNumber).asBytes(), h256::AlignRight));
```

At some point we may be able to fix this whole system not to have workarounds, but I'd say it is not worth it for the moment.

**Review by axic — APPROVED** [review]

File: test/libsolidity/SemanticTest.cpp [hunk 4]
```diff
@@ -151,9 +151,19 @@ map<string, Builtin> SemanticTest::makeBuiltins() const
...
+		},
+		{
+			"account",
+			[this](FunctionCall const& _call) -> optional<bytes>
+			{
+				soltestAssert(_call.arguments.parameters.size() == 1, "Account number expected.");
+				size_t accountNumber = static_cast<size_t>(stoi(_call.arguments.parameters.at(0).rawString));
+				// Need to pad it to 32-bytes to workaround limitations in BytesUtils::formatHex.
+				return toBigEndian(h256(ExecutionFramework::setAccount(accountNumber).asBytes(), h256::AlignRight));
+			}
```
Comment (cameel) [inline test/libsolidity/SemanticTest.cpp:165]: I don't see anything here that would require dropping the `const` from `makeBuiltins()`. Why was that necessary? The creation of builtins does not seem like an operation that would require modifying anything in `SemanticTest`.
